crdppf/static/js/Crdppf/layerTree.js

```diff
@@ -1,209 +1,219 @@
+        for (var i=0;i<baseLayers.length;i++){
+            // default: check first layer
+            isChecked = false;
+            if (i === 0){
+                isChecked = true;
+            }
+
+            // fill tree with nodes relative to baseLayers
+            var baseLayerItem =  new Ext.tree.TreeNode({
+                text: labels[baseLayers[i].name],
+                draggable:false,
+                id: baseLayers[i].wmtsname,
+                format: baseLayers[i].tile_format,
+                leaf: true,
+                checked: isChecked,
+                cls: 'baseLayerNodeCls',
+                listeners: {
+                    'checkchange': function(node, checked){
+                        if(checked){
+                            for (var k=0; k < node.parentNode.childNodes.length; k++){
+                                if(node.parentNode.childNodes[k].id != node.id){
+                                    node.parentNode.childNodes[k].getUI().toggleCheck(false);
+                                } else {
+                                    // set new backgound layer
+                                    var theBaseLayer = Crdppf.Map.map.getLayersBy('id', 'baseLayer')[0];
+                                    
+                                    if(theBaseLayer) {
+                                        theBaseLayer.destroy();
+                                    }
+                                    var formatSuffixMap;
+                                     
+                                    if (node.attributes.format == 'image/png'){
+                                         formatSuffixMap = {'image/png':'png'};
+                                    }
+                                    if (node.attributes.format == 'image/jpg'){
+                                         formatSuffixMap = {'image/jpeg':'jpg'};
+                                    }
+                                    if (node.attributes.format == 'image/jpeg'){
+                                         formatSuffixMap = {'image/jpeg':'jpeg'};
+                                    }
+                                    var layer = new OpenLayers.Layer.WMTS({
+                                        name: "Base layer",
+                                        url: Crdppf.mapproxyUrl,
+                                        layer: node.id,
+                                        matrixSet: Crdppf.mapMatrixSet,
+                                        format: node.attributes.format,
+                                        formatSuffixMap: formatSuffixMap,
+                                        isBaseLayer: true,
+                                        style: 'default',
+                                        fixedLayer: true,
+                                        requestEncoding: 'REST'
+                                    }); 
+
+                                    layer.id = 'baseLayer';
```

Comment:
Why isn't this property directly initialized when creating the object (thus 3 line above...) ?